cmd/nerdctl/push.go

```diff
@@ -183,7 +183,7 @@ func pushAction(cmd *cobra.Command, args []string) error {
 		return err
 	}
 	if err = pushFunc(resolver); err != nil {
-		if !imgutil.IsErrHTTPResponseToHTTPSClient(err) {
+		if !imgutil.IsErrHTTPResponseToHTTPSClient(err) && !imgutil.IsThePortHasBeenClosed(err) {
```

Comment:
Could you add comment lines to explain this logic?

Reply:
Do we need this for pull too?

Reply:
Also, can we have a test in https://github.com/containerd/nerdctl/blob/master/cmd/nerdctl/push_linux_test.go ?

Reply:
I will check the `login` and `pull` actions need this PR or not. In my test `nerdctl --insecure-registry login` with this situation works correctly.

But we have already test `nerdctl push` with `--insecure-registry`. It seems a little bit hard to make a only 80 port with pure HTTP. I will try to add extra test.

Anyway I will update the PR with extra comments.